src/paperless/settings.py

```diff
@@ -317,6 +319,15 @@ def __get_boolean(key, default="NO"):
 DATE_ORDER = os.getenv("PAPERLESS_DATE_ORDER", "DMY")
 FILENAME_DATE_ORDER = os.getenv("PAPERLESS_FILENAME_DATE_ORDER")
 
+# Transformations applied before filename parsing
+FILENAME_PARSE_TRANSFORMS = []
+_filename_parse_transforms = os.getenv("PAPERLESS_FILENAME_PARSE_TRANSFORMS")
+if _filename_parse_transforms:
+    FILENAME_PARSE_TRANSFORMS = [(
+            re.compile(t["pattern"], flags = re.IGNORECASE), t["repl"])
```

Comment:
I'm not sure if ignoring case should be the default given it can be activated through the in-line flag `(?i)` but not de-activated through such a flag. I guess people using regular expressions probably know just enough to understand they are case-sensitive by default, although I understand where this is coming from and I'm not strictly opposed.

Either way, it should definitely be documented! Can you add a note about this to the guesswork page?

Reply:
I'll remove ignorecase here, usually devices only emit one format so one hardly would need case insensitive matches anyway.